### PR TITLE
Add TempAppSettingsAttribute for all tests

### PR DESCRIPTION
**Pull request by mstv**
## Proposed changes

- Add `TempAppSettingsAttribute` for all tests
  in order to set `AppSettings.CheckForUpdates = false`

There is no need to tinker the `AppSettings.SettingsFilePath` because the user's GE settings file is not used but one which is specific for the test runner, e.g. 
`SettingsFilePath: C:\Users\appveyor\AppData\Roaming\NUnit Software\NUnit Engine\GitExtensions.settings`

## Test environment(s) 

- AppVeyor
- Windows 10

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by mstv** [comment]
This PR is ready.

BTW: After a force-push, it does not suffice to reload the PR in the browser once: Each of the tabs, especially "Conversation" and "Files changed", must be reloaded individually.

**Comment by mstv** [comment]
The test runner uses `C:\Users\appveyor\AppData\Roaming\NUnit Software\NUnit Engine\GitExtensions.settings` on AppVeyor.

**Comment by RussKie** [comment]
What about when tests run locally?
I just want to confirm we won't corrupt the installed app's config.

**Comment by mstv** [comment]
It is similar when run in MSVS2019:
`AppSettings.SettingsFilePath = "C:\\Users\\<user>\\AppData\\Roaming\\Microsoft Corporation\\Microsoft.TestHost.x86\\GitExtensions.settings"`.

**Comment by RussKie** [comment]
All good then, thank you

File: UnitTests/CommonTestUtils/TempAppSettingsAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using GitCommands;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+
+namespace CommonTestUtils
+{
+    [AttributeUsage(AttributeTargets.Assembly)]
+    public sealed class TempAppSettingsAttribute : Attribute, ITestAction
```
Comment (RussKie) [inline UnitTests/CommonTestUtils/TempAppSettingsAttribute.cs]: Maybe?
```suggestion
    public sealed class TestAppSettingsAttribute : Attribute, ITestAction
```

Comment (mstv) [reply UnitTests/CommonTestUtils/TempAppSettingsAttribute.cs]: Yes, will rename. The name `TempAppSettingsAttribute` was chosen with the expectation that the `AppSettings.SettingsFilePath` needed to be tinkered.

**Comment by mstv** [comment]
> Can this be used to run tests in parallel too?

No, `AppSettings` is still a singleton.
This one suppresses only the update popup.

**Comment by mstv** [comment]
@msftbot merge in 1 day

**Pull request merged**